Update location-verification.yaml for Spring25

jlurien opened the pull request:
#### What type of PR is this?

* enhancement/feature

#### What this PR does / why we need it:

Updates location-verification API with the changes derived from Commonalities and ICM updates
x-camara-guidelines: 0.5.0

#### Which issue(s) this PR fixes:

Fixes #276, #277 

#### Special notes for reviewers:

There are still some open discussions in Commonalities that may impact this PR, so we will keep it open until there is a final version for x-camara-guidelines: 0.5.0

#### Changelog input

```
* Error schemas updated with enums
* Some error status and codes have been updated
* Section with guidelines about how to identify the device from access token and request body has been updated 
```

#### Additional documentation

alpaycetin74 commented on code/API_definitions/location-verification.yaml:
```diff
@@ -53,29 +53,19 @@ info:
-    This specification defines the `device` object field as optional in API requests, specifically in cases where the API is accessed using a 3-legged access token, and the device can be uniquely identified by the token. This approach simplifies API usage for API consumers by relying on the device information associated with the access token used to invoke the API.
+    This API requires the API consumer to identify a device as the subject of the API as follows:
+    - When the API is invoked using a two-legged access token, the subject will be identified from the optional `device` object, which therefore MUST be provided.
+    - When a three-legged access token is used however, this optional identifier MUST NOT be provided, as the subject will be uniquely identified from the access token.
```
If both the 3-legged token and the device is provided, will the request be simply rejected? Is it possible to let the request continue if the token and device match?

alpaycetin74 commented on code/API_definitions/location-verification.yaml:
```diff
@@ -489,17 +484,25 @@ components:
                 message: Request not authenticated due to missing, invalid, or expired credentials.
             GENERIC_401_AUTHENTICATION_REQUIRED:
-              summary: Generic Authentication Required
               description: New authentication is needed, authentication is no longer valid
```
is it possible to explain the difference between "authentication is no longer valid" and "expired credentials" in more detail ?

jlurien replied:
There is an issue in Commonalities to discuss this: https://github.com/camaraproject/Commonalities/issues/368

IMO, UNAUTHENTICATED is the most generic one, which would cover all cases where the access token is not longer valid (typically because it is expired). AUTHENTICATION_REQUIRED is more specific, when a full new authentication from scratch is needed. But we could use the generic one and rely on the message for further description.

bigludo7 reviewed (dismissed):
LGTM

jlurien reviewed (commented):
- Test plan should be updated in the same PR

jlurien commented:
@bigludo7 @maxl2287 

I have updated the PR with new changes:

* Updated the test plan. For that I have copied the **fixed** version of the test plan artifact in Commonalities, now still proposed as [PR](https://github.com/camaraproject/Commonalities/pull/381), but likely to be merged. We would probably need to apply as well to location-retrieval and check what applies to geofencing

* From the spec, I have already removed 403 INVALID_TOKEN_CONTEXT, and 401 AUTHENTICATION_REQUIRED. Discussions around this are not fully closed but it seems this will be the consensus.

bigludo7 commented on code/Test_definitions/location-verification.feature:
```diff
@@ -115,70 +117,82 @@ Feature: CAMARA Device location verification API, v1.0.0 - Operation verifyLocat
+  Scenario: None of the provided device identifiers is supported by the implementation
+    Given that some types of device identifiers are not supported by the implementation
+    And the request body property "$.device" only includes device identifiers not supported by the implementation
+    When the HTTP "POST" request is sent
```
We should probably add: And the header "Authorization" is set to a valid access token which does not identify a single device

jlurien replied:
Yes, you are right, but this is copied from Commonalities artifact, so I will have to update there as well

bigludo7 reviewed (commented):
One small comment on the test definition else good for me.

bigludo7 reviewed (approved):
LGTM
